### DISCUSSION TITLE
Understanding KaTeX's Handling of Math Sizes (Why is it the way it is?)

### DISCUSSION THREAD

**Discussion by JacobBeaudry** [KaTeX/KaTeX#3543] [Q&A]
I've been using the `\small`, `\tiny`, `\scriptsize` _etc_ commands in LaTeX for a long time -- unaware that they aren't actually supported in math mode (because, apparently, they accidentally work in various situations which must've overlapped with my use), so I didn't think twice when I saw that KaTeX uses these commands for resizing math. 

However, a recent issue led me to research these commands & every LaTeX reference I find, including the tex.SE, says that these commands aren't meant for use in math mode. This made me curious to know why these commands have been implemented if they are not (meant to be) supported on LaTeX? and what is the recommended way of handling font-size changes with KaTeX?

Also: are these commands meant to be used as switches or macros in KaTeX? _i.e._  should it be `\small \sin^2\beta + \cos^2 \beta`, `\small{\sin^2\alpha + \cos^2\alpha}` or `{\small \sin^2 \gamma + \cos^2 \gamma}` ? I've run into an issue with a custom tooltip element undoing these size commands and thought it may be because I'm not using them as intended.

FWIW -- it doesn't seem like there is a single recommended method for handling different font-sizes in math content in LaTeX, but a recurring suggesting is to use`\scalebox` (which IINM is not supported on KaTeX, ATM).

**Comment by edemaine** [comment]
This decision predates me joining the project.  One guess is that the decision was for compatibility with MathJax (at least, the version at the time).

I agree that it's not ideal &mdash; perhaps `strict: 'warn'` mode should warn that these aren't supported in LaTeX, so others aren't surprised by this when porting KaTeX code to LaTeX.

Part of the limitation here is that KaTeX is given only math mode content, not text mode content (#604). In LaTeX, if you use `\small` in text mode, then the subsequent math mode font size *does* change (probably in the same way that KaTeX resizes).  So `\small $x^2$` in LaTeX probably behaves like `\small x^2` in KaTeX.  Maybe a solution to #604 could try to address this, or we could add a separate `fontSize` option to `render`/`renderToString` which defines the surrounding font size.
